modin/numpy/arr.py

```diff
@@ -190,6 +205,38 @@ def __init__(
             {col_name: new_dtype for col_name in self._query_compiler.columns}
         )
 
+    def _add_sibling(self, sibling):
+        """
+        Add an array object to the list of siblings.
+
+        Siblings are objects that share the same query compiler. This function is called
+        when a shallow copy is made.
+
+        Parameters
+        ----------
+        sibling : BasePandasDataset
+            Dataset to add to siblings list.
+        """
+        sibling._siblings = self._siblings + [self]
+        self._siblings += [sibling]
+        for sib in self._siblings:
+            sib._siblings += [sibling]
+
+    def _update_inplace(self, new_query_compiler):
```

Comment:
same goes for this function, it's exact copy of the one in `BasePandasDataset` - I would propose to extract the `._siblings` and these methods into a [common mixin](https://www.pythontutorial.net/python-oop/python-mixin/)